src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1435,6 +1435,12 @@ public static bool IsInstanceContext(this SyntaxTree syntaxTree, int position, S
                 return IsInBlock(accessor.Body, position);
             }
 
+            var propertyDeclaration = containingMember as PropertyDeclarationSyntax;
+            if (propertyDeclaration != null)
+            {
+                return isAnyExpressionContext && propertyDeclaration.ExpressionBody.IsKind(SyntaxKind.ArrowExpressionClause);
```

Comment:
You can just check if you're inside the ExpressionBody of a Property/Method declaration.